src/v/cloud_storage/probe.cc

```diff
@@ -12,12 +12,15 @@
 
 #include "prometheus/prometheus_sanitize.h"
 
+#include "ssx/metrics.h"
+
 #include <seastar/core/metrics.hh>
 #include <seastar/core/smp.hh>
 
 namespace cloud_storage {
 
-remote_probe::remote_probe(remote_metrics_disabled disabled) {
+remote_probe::remote_probe(remote_metrics_disabled disabled)
+  : _public_metrics(ssx::public_metrics_handle) {
     if (disabled) {
```

Comment:
There's another flag for `disable_public_metrics`, so I think the early return is wrong.

Reply:
disable_public_metrics comes from v/config/configuration where as the cloud storage subsystem uses the configuration from v/net/transport .

I'll have to scan through the code base more to see how to reconcile this difference.

Reply:
Done. I had to add public metrics to the configuration container in src/v/net

Changes at https://github.com/redpanda-data/redpanda/pull/5367